### PR TITLE
chore: bump anchor v1

### PR DESCRIPTION
**Pull request by codewithgun**

### CODE REVIEW

File: .github/workflows/ci_rs_main_on_push.yml [hunk 1]
```diff
@@ -40,33 +40,31 @@ jobs:
...
       # Testing
-      - run: npm install
+      - uses: oven-sh/setup-bun@v2
+      - run: bun install
         shell: bash
-      - run: cd programs && cargo test -- --nocapture && cargo test-bpf -- --nocapture && cd ../..
+      - run: cd programs && cargo test -- --nocapture && cd ../..
         shell: bash
-      - run: anchor test -- --features devnet
+      - run: anchor test --ignore-keys -- --features devnet
```
Comment (bangyro) [inline .github/workflows/ci_rs_main_on_push.yml]: `--ignore-keys` doesn't work with anchor test.
there's no test defined in `Anchor.toml` 
and I think both of the CI actions are disabled manually, which is why there's no ci runs

File: programs/farming/src/lib.rs [hunk 2]
```diff
@@ -807,10 +803,10 @@ pub struct Fund<'info> {
     )]
     funder: Signer<'info>,
     /// Funder reward A ATA
-    #[account(mut)]
+    #[account(mut, dup)]
```
Comment (bangyro) [inline programs/farming/src/lib.rs:806]: I think we need `account(mut, dup)` constraint on the `ClosePool's` `reward_a_refundee` and `reward_b_refundee` account for the case where there is only 1 reward mint (i.e. `reward_a_mint` and `reward_b_mint` is the same).

I don't think we have a constraint where `staking_mint != reward_mint` so its possible `staking_refundee` also need the `dup`.

File: tests/farming/reward_precision.ts [hunk 3]
```diff
@@ -1,776 +1,356 @@
...
-        .signers([u.keypair])
-        .rpc()
-    )
-  );
+function mintToMultiple(svm: LiteSVM, payer: Keypair, mint: PublicKey, mintAuthority: Keypair, atas: PublicKey[]) {
+  for (const ata of atas) {
+    // mint directly to ATA (not to wallet)
+    const owner = svm.getAccount(ata);
+    // Actually we need to use the raw instruction approach
+    const { createMintToInstruction } = require("@solana/spl-token");
```
Comment (bangyro) [inline tests/farming/reward_precision.ts]: why not use import?

File: tests/farming/reward_precision.ts [hunk 4]
```diff
@@ -1,776 +1,356 @@
...
-          user: u.user,
-        })
-        .signers([u.keypair])
-        .rpc()
-    )
-  );
+function mintToMultiple(svm: LiteSVM, payer: Keypair, mint: PublicKey, mintAuthority: Keypair, atas: PublicKey[]) {
+  for (const ata of atas) {
+    // mint directly to ATA (not to wallet)
+    const owner = svm.getAccount(ata);
```
Comment (bangyro) [inline tests/farming/reward_precision.ts]: owner is unused

File: tests/farming/reward_precision.ts [hunk 5]
```diff
@@ -1,776 +1,356 @@
...
-const program = anchor.workspace.Farming as Program<Farming>;
-const BASE_KEYPAIR = Keypair.generate();
-const ADMIN_KEYPAIR = Keypair.generate();
-const USER_KEYPAIR = Keypair.generate();
+import {
+  startSvm,
+  createFarmingProgram,
+  createToken,
+  getOrCreateAssociatedTokenAccount,
+  mintSplTokenTo,
```
Comment (bangyro) [inline tests/farming/reward_precision.ts]: mintSplTokenTo is unused

File: tests/farming/dual-farming-edge-case.ts [hunk 6]
```diff
@@ -15,773 +10,256 @@ import {
...
-      stakingVaultAddress.toBase58()
-    );
-    assert.deepStrictEqual(
-      poolState.rewardAVault.toBase58(),
-      rewardAVaultAddress.toBase58()
-    );
-    assert.deepStrictEqual(
-      poolState.rewardBVault.toBase58(),
-      rewardBVaultAddress.toBase58()
-    );
```
Comment (bangyro) [inline tests/farming/dual-farming-edge-case.ts:274]: why did we remove the checks on the pool state?

File: package.json [hunk 7]
```diff
@@ -1,16 +1,20 @@
...
+        "@anchor-lang/core": "^1.0.0",
+        "@solana/spl-token": "^0.4.13",
+        "@solana/web3.js": "^1.95.0",
         "ts-mocha": "^10.0.0",
-        "ts-node": "^10.4.0",
-        "typescript": "^4.5.4"
+        "typescript": "^5.0.0"
     },
     "devDependencies": {
+        "@types/bn.js": "^5.1.0",
```
[the remark this answers is not in the document]
Comment (codewithgun) [reply package.json:16]: Ignored this since the test is working

File: .github/workflows/ci_rs_main_on_push.yml (hunk 1 above)
Comment (codewithgun) [reply .github/workflows/ci_rs_main_on_push.yml]: Fixed

File: programs/farming/src/lib.rs (hunk 2 above)
Comment (codewithgun) [reply programs/farming/src/lib.rs:806]: Fixed

File: tests/farming/reward_precision.ts (hunk 3 above)
Comment (codewithgun) [reply tests/farming/reward_precision.ts]: Fixed

File: tests/farming/reward_precision.ts (hunk 4 above)
Comment (codewithgun) [reply tests/farming/reward_precision.ts]: Fixed

File: tests/farming/reward_precision.ts (hunk 5 above)
Comment (codewithgun) [reply tests/farming/reward_precision.ts]: Fixed

File: tests/farming/dual-farming-edge-case.ts (hunk 6 above)
Comment (codewithgun) [reply tests/farming/dual-farming-edge-case.ts:274]: Reverted

File: .github/workflows/ci_ts_main_on_push.yml
Comment (bangyro) [inline .github/workflows/ci_ts_main_on_push.yml:1]: Can delete this file. Since `.github/workflows/ci_rs_main_on_push.yml` now runs the ts test.

Comment (codewithgun) [reply .github/workflows/ci_ts_main_on_push.yml:1]: Fixed

**Review by bangyro — APPROVED** [review]
